test/unit/math/mathOps/include/Functor.hpp

```diff
@@ -0,0 +1,288 @@
+                 std::is_same< TAcc, std::nullptr_t>::value,                    \
+                 int>::type = 0>                                                \
+    ALPAKA_FN_HOST                                                              \
+    auto execute(                                                               \
+        TAcc const & acc,                                                       \
+        TArgs const &... args ) const                                           \
+    -> decltype( STD_OP( args... ) )                                            \
+    {                                                                           \
+      alpaka::ignore_unused( acc );                                             \
+      return STD_OP( args... );                                                 \
+    }                                                                           \
+                                                                                \
+    /* assigns args by arity */                                                 \
+    ALPAKA_NO_HOST_ACC_WARNING                                                  \
+    template<                                                                   \
+        typename T,                                                             \
+        typename TAcc = std::nullptr_t>                                         \
+    ALPAKA_FN_HOST_ACC                                                          \
+    auto operator()(                                                            \
+        ArgsItem<T, Arity::Unary> const & args,                                 \
+        TAcc const & acc = nullptr) const                                       \
+    -> decltype(execute(acc, args.arg[0]))                                      \
+    {                                                                           \
+        return execute(acc, args.arg[0]);                                       \
+    }                                                                           \
+                                                                                \
+    /* assigns args by arity */                                                 \
+    ALPAKA_NO_HOST_ACC_WARNING                                                  \
+    template<                                                                   \
+        typename T,                                                             \
+        typename TAcc = std::nullptr_t>                                         \
+    ALPAKA_FN_HOST_ACC                                                          \
+    auto operator()(                                                            \
+        ArgsItem<T, Arity::Binary> const & args,                                \
+        TAcc const & acc = nullptr) const                                       \
+    -> decltype(execute(acc, args.arg[0], args.arg[1]))                         \
```

Comment:
Now that we have C++14 as baseline, we can remove this line.

Reply:
It could be that the return type is required because of CUDA restrictions. We need to test it here.